tests/notebooks/e2e-wine/e2e-wine-kfp-mlflow-seldon.ipynb

```diff
@@ -0,0 +1,1233 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "e89bf654",
+   "metadata": {},
+   "source": [
+    "# E2E scenario for Wine dataset on KFP\n",
+    "\n",
+    "Steps:\n",
+    "- download dataset\n",
+    "- clean/preprocess data\n",
+    "- perform training / hyper-parameter tuning with results in MLFlow + MinIO\n",
+    "- serve with Seldon\n",
+    "- perform inference\n",
+    "\n",
+    "Artifacts:\n",
+    "- raw data, preprocessed\n",
+    "- model per experiment\n",
+    "- experiment metadata and results"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "3ce43dce-51b6-4d90-86d2-be1dfafff9a8",
+   "metadata": {},
+   "source": [
+    "## Setup"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "id": "6e34b594",
+   "metadata": {
+    "execution": {
+     "iopub.execute_input": "2023-08-09T14:52:27.765768Z",
+     "iopub.status.busy": "2023-08-09T14:52:27.765452Z",
+     "iopub.status.idle": "2023-08-09T14:52:33.932762Z",
+     "shell.execute_reply": "2023-08-09T14:52:33.928075Z"
+    },
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "!pip install boto3 kfp==1.8.22 minio mlflow==2.1.1 numpy pyarrow requests \"scikit-learn<1.2\" tenacity -q"
```

Comment:
Was `kfp 2.0.x` not working for you?

With KFP 2.0 GA coming very soon, I think we probably want to use the latest SDK.

Reply:
Also, I think that is totally worth it to have this in a requirements.txt file if we are going to pin several packages. I think it would be great if we add some comments justifying the pin so we know when to upgrade them.

Reply:
I haven't tried with KFP 2.0, didn't know we support it, but I can test it!

Reply:
I want to have all directly installed packages specified in a requirements.in file and have everything pinned in a requirements.txt

I was planning to do this for all of the charms next week.

The packages currently pinned were just the ones needed for the notebook to work as expected!

Reply:
+1 on the requirements.in file.

Reply:
> I haven't tried with KFP 2.0, didn't know we support it, but I can test it!

It should work and it will avoid a very soon upgrade of this notebook.

Reply:
@DnPlas, unfortunately KFP 2.0 won't work in our current deployments. Trying to initialise the Client we get this warning:
```
/opt/conda/lib/python3.8/site-packages/kfp/client/client.py:158: FutureWarning: This client only works with Kubeflow Pipeline v2.0.0-beta.2 and later versions.
```
Initialisation then proceeds to fail with a scary error, not being able to find the health checks endpoint. I'll stick with 1.8 for now. When KFP 2.0 comes (as part of 1.8 I expect), we'll have a new version (i.e. branch) of these tests, which are in any case only tested on 1.7 for the time being.

Reply:
On top of that, the KFP 2.0 SDK brings other breaking changes (a lot of stuff moved to `kfp.deprecated`, e.g. the `use_k8s_secret` helper that I was using)